server/webex/client.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package webex
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+const StatusStarted = "STARTED"
+
+type ClientError struct {
+	StatusCode int
+	Err        error
+}
+
+func (ce *ClientError) Error() string {
+	return ce.Err.Error()
+}
+
+type Client interface {
+	GetPersonalMeetingRoomUrl(roomId, username, email string) (string, *ClientError)
```

Comment:
2/5 let's not return custom error types, at least not from public interfaces. Return an `error`?